wagtail/contrib/frontend_cache/backends/base.py

```diff
@@ -21,8 +21,55 @@ def purge_batch(self, urls) -> None:
         for url in urls:
             self.purge(url)
 
+    def purge_hostname(self, hostname) -> None:
+        """
+        Purge all cached URLs matching the provided ``hostname``. Where the underlying
+        service or technology supports it, specific backends can implement this to allow
+        efficient mass-purging with a minumal number of requests.
+
+        Where not implemented, ``wagtail.contrib.frontend_cache.utils.purge_site()``
+        will use ``purge_batch()`` to purge known URLs instead.
+        """
+        raise NotImplementedError
+
+    def purge_everything(self) -> None:
```

Comment:
Nit: Would `purge_all` be a better name?

Reply:
Implemented in e15760e